npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -483,17 +498,21 @@ def lockfiles_for_path(lockfiles:, path:)
         end
 
         def run_checker(path:, version:)
-          # If there are both yarn lockfiles and npm lockfiles only run the
-          # yarn updater
           yarn_lockfiles = lockfiles_for_path(lockfiles: dependency_files_builder.yarn_locks, path: path)
           return run_yarn_checker(path: path, version: version, lockfile: yarn_lockfiles.first) if yarn_lockfiles.any?
 
+          pnpm_lockfiles = lockfiles_for_path(lockfiles: dependency_files_builder.pnpm_locks, path: path)
+          return run_pnpm_checker(path: path, version: version) if pnpm_lockfiles.any?
```

Comment:
Why does yarn win over pnpm if both are present? Why not check for presence of all three, and if more than one found, raise an error that we aren't sure what we should do?

There may be a good logical reason to still do what we do today, but if so, a code comment would be helpful to explain why.

Reply:
Yeah, same conversation as above. I think we should simplify this. I just tried to do something that follows the current pattern, but I think I also failed to make it consistent anyways.

I'm going to dedicate a small amount of time to try clean this up.

Reply:
I'm cool with also just filing an issue, cross-linking it to our upcoming "ecosystem clarity" effort and moving on for now.

The other way to solve this long term is if we get the user to explicitly tell us which package manager they want us to use... `pnpm`, `yarn`, `npm`, rather than us guessing... I prefer that, although we'll need to figure out how that works in the context of security updates where a repo may not have a `dependabot.yaml` config file.

But if there's easy enough cleanup/tidying where it clarifies the code without getting all the way into redoing our config files, then yeah, that makes sense too... I trust your judgement.

Reply:
I opened a ticket! https://github.com/dependabot/dependabot-core/issues/7184